javaslang/src/main/java/javaslang/collection/Multimaps.java

```diff
@@ -0,0 +1,171 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2016 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Tuple;
+import javaslang.Tuple2;
+import javaslang.collection.Multimap.ContainerType;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import java.util.stream.Collector;
+
+/**
+ * Implementations of common {@code Map} functions (not intended to be public).
+ *
+ * @author Ruslan Sennov, Daniel Dietrich
+ * @since 2.1.0
+ */
+final class Multimaps {
+
+    static <K, V, K2, V2, M extends Multimap<K2, V2>> M bimap(
+            Multimap<K, V> multimap,
+            OfEntries<K2, V2, M> ofEntries,
+            Function<? super K, ? extends K2> keyMapper,
+            Function<? super V, ? extends V2> valueMapper) {
+        Objects.requireNonNull(keyMapper, "keyMapper is null");
+        Objects.requireNonNull(valueMapper, "valueMapper is null");
+        final Iterator<Tuple2<K2, V2>> entries = multimap.iterator().map(entry -> Tuple.of(keyMapper.apply(entry._1), valueMapper.apply(entry._2)));
+        return ofEntries.apply(entries);
+    }
+
+    static <K, V> int size(Map<K, Traversable<V>> back) {
+        return back.foldLeft(0, (s, t) -> s + t._2.size());
+    }
+
+    static <K, V> java.util.Map<K, Collection<V>> toJavaMap(Multimap<K, V> multimap) {
+        final java.util.Map<K, Collection<V>> javaMap = new java.util.HashMap<>();
+        final Supplier<Collection<V>> javaContainerSupplier;
+        final ContainerType containerType = multimap.getContainerType();
```

Comment:
you could change this to an switch instead, as `ContainerType` is an enum and maybe extract to separate method, to get rid of `break` statements, e.g.:
```java
@Override
public java.util.Map<K, Collection<V>> toJavaMap() {
    final java.util.Map<K, Collection<V>> javaMap = new java.util.HashMap<>();
    final Supplier<Collection<V>> supplier = getJavaContainerSupplier();
    for (Tuple2<K, V> t : this) {
        javaMap.computeIfAbsent(t._1, k -> supplier.get()).add(t._2);
    }
    return javaMap;
}
private Supplier<Collection<V>> getJavaContainerSupplier() {
    switch (containerType) {
        case SEQ:
            return java.util.ArrayList::new;
        case SET:
            return java.util.HashSet::new;
        case SORTED_SET:
            return java.util.TreeSet::new;
        default:
            throw new IllegalStateException("Unknown ContainerType: " + containerType);
    }
}
```